Source/SeaInk.Core/Models/Tables/SheetIndex.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Google.Apis.Sheets.v4.Data;
+
+namespace SeaInk.Core.Models.Tables
+{
+    public class SheetIndex
+    {
+        public string SheetName { get; set; }
+        public int SheetId { get; set; }
+
+        public SheetIndex(string sheetName, int sheetId)
+        {
+            SheetName = sheetName;
+            SheetId = sheetId;
+        }
+
+        protected bool Equals(SheetIndex other)
+        {
+            return SheetName == other.SheetName && SheetId == other.SheetId;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj != null && Equals((SheetIndex) obj);
+        }
```

Comment:
Выглядит не очень, попробуй так
```suggestion
        public override bool Equals(object obj)
        {
            var sheetIndex = obj as SheetIndex;
            if (sheetIndex == null)
            {
                return false;
            }
            
            return SheetName == sheetIndex.SheetName && SheetId == sheetIndex.SheetId;
        }
```

Reply:
done

Reply:
@annchous Лол, что за странное мувы?
1) Ты зачем удалила типизированный Equals?
2) Ну новый метод брух, чо так много строк
```C#
public override bool Equals(object obj)
        {
            return obj is SheetIndex sheet && Equals(sheet);
        }
```
ииии...это все!

Reply:
@hrrrrustic спасибо, учитель